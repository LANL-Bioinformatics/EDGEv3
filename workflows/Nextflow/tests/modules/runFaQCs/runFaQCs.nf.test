nextflow_workflow {

    name "Test Workflow FAQCS"
    script "modules/runFaQCs/runFaQCs.nf"
    workflow "FAQCS"

    test("Basic PE") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testFaQCs"
                    cpus = 4
                    pairedFile = true
                }
                faqcs {
                    //all defaults
                }
            }
            workflow {
                """
                input[0] = params.faqcs.plus(params.shared)
                input[1] = "ILLUMINA"
                input[2] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[3] = file("${projectDir}/nf_assets/NO_FILE2")
                input[4] = channel.of("151")
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(
                    workflow.out.qcStats,
                    workflow.out.paired,
                    workflow.out.unpaired
                    ).match()},
                {assert file(workflow.out.qcReport.find { file(it).name == "QC_qc_report.pdf" }).exists()}
            )
        }

    }

    test("Basic SE") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testFaQCs"
                    cpus = 4
                    pairedFile = false
                }
                faqcs {
                    //all defaults
                }
            }
            workflow {
                """
                input[0] = params.faqcs.plus(params.shared)
                input[1] = "ILLUMINA"
                input[2] = [file("${projectDir}/nf_assets/NO_FILE")]
                input[3] = file("${projectDir}/test_data/Ecoli_10x.1.fastq")
                input[4] = channel.of("151")
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(
                    workflow.out.qcStats,
                    workflow.out.paired,
                    workflow.out.unpaired
                    ).match()},
                {assert file(workflow.out.qcReport.find { file(it).name == "QC_qc_report.pdf" }).exists()}
            )
        }

    }

    test("PE with adapter removal") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testFaQCs"
                    cpus = 4
                    pairedFile = true
                }
                faqcs {
                    artifactFile = "${projectDir}/test_data/adapters.fasta"
                }
            }
            workflow {
                """
                input[0] = params.faqcs.plus(params.shared)
                input[1] = "ILLUMINA"
                input[2] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[3] = file("${projectDir}/nf_assets/NO_FILE2")
                input[4] = channel.of("151")
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(
                    workflow.out.qcStats,
                    workflow.out.paired,
                    workflow.out.unpaired
                    ).match()},
                {assert file(workflow.out.qcReport.find { file(it).name == "QC_qc_report.pdf" }).exists()}
            )
        }

    }

    test("SE with adapter removal") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testFaQCs"
                    cpus = 4
                    pairedFile = false
                }
                faqcs {
                    artifactFile = "${projectDir}/test_data/adapters.fasta"
                }
            }
            workflow {
                """
                input[0] = params.faqcs.plus(params.shared)
                input[1] = "ILLUMINA"
                input[2] = [file("${projectDir}/nf_assets/NO_FILE")]
                input[3] = file("${projectDir}/test_data/Ecoli_10x.1.fastq")
                input[4] = channel.of("151")
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(
                    workflow.out.qcStats,
                    workflow.out.paired,
                    workflow.out.unpaired
                    ).match()},
                {assert file(workflow.out.qcReport.find { file(it).name == "QC_qc_report.pdf" }).exists()}
            )
        }

    }

    test("PolyA removal") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testFaQCs"
                    cpus = 4
                    pairedFile = true
                }
                faqcs {
                    polyA = true
                }
            }
            workflow {
                """
                input[0] = params.faqcs.plus(params.shared)
                input[1] = "ILLUMINA"
                input[2] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[3] = file("${projectDir}/nf_assets/NO_FILE2")
                input[4] = channel.of("151")
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(
                    workflow.out.qcStats,
                    workflow.out.paired,
                    workflow.out.unpaired
                    ).match()},
                {assert file(workflow.out.qcReport.find { file(it).name == "QC_qc_report.pdf" }).exists()}
            )
        }
    }
}
