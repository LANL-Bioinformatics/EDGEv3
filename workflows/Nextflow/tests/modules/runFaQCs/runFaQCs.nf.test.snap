{
    "PolyA removal": {
        "content": [
            {
                "0": [
                    [
                        "QC.1.trimmed.fastq:md5,deb773f1726c6134a7d2fc069af7b21b",
                        "QC.2.trimmed.fastq:md5,a793e8ccfe60c0f9462a80d520eecd7a"
                    ]
                ],
                "1": [
                    "QC.unpaired.trimmed.fastq:md5,5ed2596cae072d61e24a96d3e4449611"
                ],
                "2": [
                    "QC_qc_report.pdf:md5,909b3cb4f2e3c08e481d5525c58b9947"
                ],
                "3": [
                    "QC.stats.txt:md5,29b8b7c8ea6f7b64e1bcbe9adb81ecf8"
                ],
                "paired": [
                    [
                        "QC.1.trimmed.fastq:md5,deb773f1726c6134a7d2fc069af7b21b",
                        "QC.2.trimmed.fastq:md5,a793e8ccfe60c0f9462a80d520eecd7a"
                    ]
                ],
                "qcReport": [
                    "QC_qc_report.pdf:md5,909b3cb4f2e3c08e481d5525c58b9947"
                ],
                "qcStats": [
                    "QC.stats.txt:md5,29b8b7c8ea6f7b64e1bcbe9adb81ecf8"
                ],
                "unpaired": [
                    "QC.unpaired.trimmed.fastq:md5,5ed2596cae072d61e24a96d3e4449611"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-18T15:22:51.321855314"
    },
    "Basic SE": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "QC.unpaired.trimmed.fastq:md5,3a0138c06e9cd504e2cca8e5d93532f1"
                ],
                "2": [
                    "QC_qc_report.pdf:md5,1cc30148bbe2cf222f28de748b1023a1"
                ],
                "3": [
                    "QC.stats.txt:md5,e74be317e4f05f00ef89b4c29487cc6b"
                ],
                "paired": [
                    
                ],
                "qcReport": [
                    "QC_qc_report.pdf:md5,1cc30148bbe2cf222f28de748b1023a1"
                ],
                "qcStats": [
                    "QC.stats.txt:md5,e74be317e4f05f00ef89b4c29487cc6b"
                ],
                "unpaired": [
                    "QC.unpaired.trimmed.fastq:md5,3a0138c06e9cd504e2cca8e5d93532f1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-18T15:21:48.733823563"
    },
    "SE with adapter removal": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "QC.unpaired.trimmed.fastq:md5,3e43d37648eea875f5583f5cfcbb16ca"
                ],
                "2": [
                    "QC_qc_report.pdf:md5,92f082c1846aff7b4d5b55e30d7c16bb"
                ],
                "3": [
                    "QC.stats.txt:md5,4b6b872f59511bf1db31de374018456c"
                ],
                "paired": [
                    
                ],
                "qcReport": [
                    "QC_qc_report.pdf:md5,92f082c1846aff7b4d5b55e30d7c16bb"
                ],
                "qcStats": [
                    "QC.stats.txt:md5,4b6b872f59511bf1db31de374018456c"
                ],
                "unpaired": [
                    "QC.unpaired.trimmed.fastq:md5,3e43d37648eea875f5583f5cfcbb16ca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-18T15:22:35.167592113"
    },
    "Basic PE": {
        "content": [
            {
                "0": [
                    [
                        "QC.1.trimmed.fastq:md5,deb773f1726c6134a7d2fc069af7b21b",
                        "QC.2.trimmed.fastq:md5,a793e8ccfe60c0f9462a80d520eecd7a"
                    ]
                ],
                "1": [
                    "QC.unpaired.trimmed.fastq:md5,5ed2596cae072d61e24a96d3e4449611"
                ],
                "2": [
                    "QC_qc_report.pdf:md5,7d600c011a7c4a73b232b88f9d264b4c"
                ],
                "3": [
                    "QC.stats.txt:md5,29b8b7c8ea6f7b64e1bcbe9adb81ecf8"
                ],
                "paired": [
                    [
                        "QC.1.trimmed.fastq:md5,deb773f1726c6134a7d2fc069af7b21b",
                        "QC.2.trimmed.fastq:md5,a793e8ccfe60c0f9462a80d520eecd7a"
                    ]
                ],
                "qcReport": [
                    "QC_qc_report.pdf:md5,7d600c011a7c4a73b232b88f9d264b4c"
                ],
                "qcStats": [
                    "QC.stats.txt:md5,29b8b7c8ea6f7b64e1bcbe9adb81ecf8"
                ],
                "unpaired": [
                    "QC.unpaired.trimmed.fastq:md5,5ed2596cae072d61e24a96d3e4449611"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-18T15:21:36.155371528"
    },
    "PE with adapter removal": {
        "content": [
            {
                "0": [
                    [
                        "QC.1.trimmed.fastq:md5,1e70223f29baaced4509b62a5e84f074",
                        "QC.2.trimmed.fastq:md5,649bf376592b80e9ae1cee81d018a65f"
                    ]
                ],
                "1": [
                    "QC.unpaired.trimmed.fastq:md5,1ca05e3fe31af0ca222651c3b5135cf7"
                ],
                "2": [
                    "QC_qc_report.pdf:md5,c5ae94def6d3a6b43d1c682b311239b3"
                ],
                "3": [
                    "QC.stats.txt:md5,7e832f626a1f59314a2c6df4d1ed7db2"
                ],
                "paired": [
                    [
                        "QC.1.trimmed.fastq:md5,1e70223f29baaced4509b62a5e84f074",
                        "QC.2.trimmed.fastq:md5,649bf376592b80e9ae1cee81d018a65f"
                    ]
                ],
                "qcReport": [
                    "QC_qc_report.pdf:md5,c5ae94def6d3a6b43d1c682b311239b3"
                ],
                "qcStats": [
                    "QC.stats.txt:md5,7e832f626a1f59314a2c6df4d1ed7db2"
                ],
                "unpaired": [
                    "QC.unpaired.trimmed.fastq:md5,1ca05e3fe31af0ca222651c3b5135cf7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-18T15:22:14.967285508"
    }
}